workers/unity/Packages/com.improbable.gdk.core/Systems/ComponentSendSystem.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Improbable.Gdk.Core.CodegenAdapters;
+using Improbable.Worker.CInterop;
+using Unity.Entities;
+using UnityEngine.Profiling;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Executes the default replication logic for each SpatialOS component.
+    /// </summary>
+    [DisableAutoCreation]
+    [AlwaysUpdateSystem]
+    [UpdateInGroup(typeof(SpatialOSSendGroup.InternalSpatialOSSendGroup))]
+    [UpdateBefore(typeof(ComponentUpdateSystem))]
+    public class ComponentSendSystem : ComponentSystem
+    {
+        // Can't access the generated component ID in Core code.
+        private const uint PositionComponentId = 54;
+
+        private Connection connection;
+
+        private readonly List<ComponentReplicator> componentReplicators =
+            new List<ComponentReplicator>();
+
+        protected override void OnCreateManager()
+        {
+            base.OnCreateManager();
+
+            connection = World.GetExistingManager<WorkerSystem>().Connection;
+
+            PopulateDefaultComponentReplicators();
+        }
+
+        public bool TryRegisterCustomReplicationSystem(uint componentId)
+        {
+            if (componentReplicators.All(componentReplicator => componentReplicator.ComponentId != componentId))
+            {
+                return false;
+            }
+
+            // The default replication system is removed, instead the custom one is responsible for replication.
+            return componentReplicators.Remove(componentReplicators.First(
+                componentReplicator => componentReplicator.ComponentId == componentId));
+        }
+
+        protected override void OnUpdate()
+        {
+            var componentUpdateSystem = World.GetExistingManager<ComponentUpdateSystem>();
```

Comment:
same as for the other system: should we move that to after the "is connection null" check

Reply:
Sure